normative/dtpr-stickers
# Chore/firestore queries

### Notion card link(s)

- [Chore: adjust firebase calls to conditionally query data by a 'wildcard' customerId if customerId isn't matched](https://www.notion.so/normative/Chore-adjust-firebase-calls-to-conditionally-query-data-by-a-wildcard-customerId-if-customerId-is-3e116aec6a354ba280aff18e4fc1ae97)

### Description (Proposed changes, additional context, chosen approach/alternative approaches, gotchas, etc)

- Adds a conditional statement to collection queries as we discussed.
- Adds optional chaining because some of the test data doesn't have all data fields (e.g. not all sensors have a systemIds array or a datachain).
- The 'wildcard' customerId is "allCustomers", this will need to be set in the Heroku config vars as the customerId for the [staging environment](https://dashboard.heroku.com/apps/dtpr-staging-normative/settings).
- I think we will still may need UI for instances where customerId isn't matched (captured in [this](https://www.notion.so/normative/Feature-UI-for-customerId-error-e178f708c814402ba9541c619f7903ba) card).

### Steps for reviewer(s) to reproduce/test

Using review app:
- In the [review app settings](https://dashboard.heroku.com/apps/dtpr-sticker-chore-fire-oo5kgr/settings) the config var for customerId had been set to "allCustomers". You should be able to open & view pages for each of these links (all have different customerIds):
[sensor 1](https://dtpr-sticker-chore-fire-oo5kgr.herokuapp.com/sensors/mULBhQ3KFJqUYtyWWpJT)
[place 1](https://dtpr-sticker-chore-fire-oo5kgr.herokuapp.com/places/bCHv97rub6Srk1uaZHMd)
[sensor 2](https://dtpr-sticker-chore-fire-oo5kgr.herokuapp.com/sensors/Jg9UXqD9cpM3zN31Evh0)
[place 2](https://dtpr-sticker-chore-fire-oo5kgr.herokuapp.com/places/xOQIMRMqwyTleqWxRJuV)

- Change the customerId config var to "XG7XJwK2bmf5PrtOJjlX" (you may need to re-deply the app) and you should only be able to open [place 1](https://dtpr-sticker-chore-fire-oo5kgr.herokuapp.com/places/bCHv97rub6Srk1uaZHMd) and [sensor 1](https://dtpr-sticker-chore-fire-oo5kgr.herokuapp.com/sensors/mULBhQ3KFJqUYtyWWpJT), but none of the other links above.

Using localhost:
- Change customerId in your .env to "all Customers" and test these links (all have different customerIds):
[sensor 1](http://localhost:3000/sensors/mULBhQ3KFJqUYtyWWpJT)
[place 1](http://localhost:3000/places/bCHv97rub6Srk1uaZHMd)
[sensor 2](http://localhost:3000/sensors/Jg9UXqD9cpM3zN31Evh0)
[place 2](http://localhost:3000/places/xOQIMRMqwyTleqWxRJuV)

- Change the customerId in .env to "XG7XJwK2bmf5PrtOJjlX" and only [sensor 1](http://localhost:3000/sensors/mULBhQ3KFJqUYtyWWpJT) & [place 1](http://localhost:3000/places/bCHv97rub6Srk1uaZHMd) links should work.
